packages/graphback-codegen-client/src/templates/gqlTemplates.ts

```diff
@@ -1,182 +1,189 @@
-    .map((f: InputModelFieldContext) => `\$${f.name}${f.isType ? 'Id' : ''}: ${f.isType ? 'Int' : f.type}${f.isNull ? '' : '!'}`)
-    .join(', ')
-}
-
-export const variables = (t: InputModelTypeContext) => {
-  return t.fields.filter((f: InputModelFieldContext) => !f.isArray)
-    .map((f: InputModelFieldContext) => `${f.name}: \$${f.name}${f.isType ? 'Id' : ''}`)
-    .join(', ')
+  return `fragment ${t.name}ExpandedField on ${t.name} {
+${returnFieldsString}
+} `
 }
 
-export const inputVariables = (t: InputModelTypeContext) => {
-  return t.fields.filter((f: InputModelFieldContext) => !f.isArray && f.type !== 'ID')
-    .map((f: InputModelFieldContext) => `${f.name}${f.isType ? 'Id' : ''}: \$${f.name}${f.isType ? 'Id' : ''}`)
-    .join(', ')
-}
-
-export const findAllQuery = (t: InputModelTypeContext) => {
+export const findAllQuery = (t: GraphQLObjectType) => {
   const fieldName = getFieldName(t.name, GraphbackOperationType.FIND_ALL, 's')
 
   return `query ${fieldName} {
     ${fieldName} {
-      ...${t.name}Fields
+      ...${t.name}ExpandedFields
     }
   }`
 }
 
-export const findQuery = (t: InputModelTypeContext) => {
+export const findQuery = (t: GraphQLObjectType) => {
   const fieldName = getFieldName(t.name, GraphbackOperationType.FIND, 's')
 
-  return `query ${fieldName}(${variableFields(t)}) {
-    ${fieldName}(fields: {${variables(t)}}) {
-      ...${t.name}Fields
+  return `query ${fieldName} ($filter: NoteFilter!)}) {
+    ${fieldName}(filter: $filter}) {
+      ...${ t.name}ExpandedFields
     }
   }`
 }
 
 
-export const createMutation = (t: InputModelTypeContext) => {
+export const createMutation = (t: GraphQLObjectType) => {
   const fieldName = getFieldName(t.name, GraphbackOperationType.CREATE)
 
-  return `mutation ${fieldName}(${inputVariableFields(t)}) {
-    ${fieldName}(input: {${inputVariables(t)}}) {
-      ...${t.name}Fields
-    }
+  return `mutation ${fieldName} ($input: ${t.name}Input!) {
+  ${ fieldName} (input: ${t.name}Input!) {
+      ...${ t.name}Fields
   }
+}
 `
 }
 
-export const updateMutation = (t: InputModelTypeContext) => {
+export const updateMutation = (t: GraphQLObjectType) => {
   const fieldName = getFieldName(t.name, GraphbackOperationType.UPDATE)
 
-  return `mutation ${fieldName}($id: ID!, ${inputVariableFields(t)}) {
-    ${fieldName}(id: $id, input: {${inputVariables(t)}}) {
-      ...${t.name}Fields
-    }
+  return `mutation ${fieldName} ($input: ${t.name}Input!) {
+  ${ fieldName} (input: ${t.name}Input!) {
+      ...${ t.name}Fields
   }
+}
 `
 }
 
-export const deleteMutation = (t: InputModelTypeContext, ) => {
+export const deleteMutation = (t: GraphQLObjectType) => {
   const fieldName = getFieldName(t.name, GraphbackOperationType.DELETE)
 
-  return `mutation ${fieldName}($id: ID!) {
-    ${fieldName}(id: $id)
+  return `mutation ${fieldName} ($input: ${t.name}Input!) {
+  ${fieldName} (input: ${t.name}Input!) {
```

Comment:
Generated schema input types do not contain an ID field. This would produce the following:

```
NoteInput
  title
  description
```

This will try to delete a non-unique item (multiple notes can have the same description and title). Same goes for createMutations.

Reply:
Yes.. This is TODO we need to add to the schema, but it depends on the ID mapping logic from the core so I left that without TODO now

Reply:
Target here is do not do half backed generation engine. The only thing outstanding is just field, relation and id mapping that will allow us to fix all other generators